### DISCUSSION TITLE
Outdated pod-identity-webhook version includes a bug that disallows injection of  AWS_STS_REGIONAL_ENDPOINTS=regional

### DISCUSSION THREAD

**Discussion by matveidner** [okd-project/okd#1686] [General]
**Describe the bug**
VPCE requires [SetRegion and SetEndpoint](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_sts_vpce.html#:~:text=After%20you%20create%20the%20VPC%20endpoint%2C%20you%20must%20use%20the%20matching%20regional%20endpoint%20to%20send%20your%20AWS%20STS%20requests.%20AWS%20STS%20recommends%20that%20you%20use%20both%20the%20setRegion%20and%20setEndpoint%20methods%20to%20make%20calls%20to%20a%20Regional%20endpoint.) parameters explicitly defined to be automatically used over the global (default) STS endpoint. The AWS-Pod-Identity-Webhook operator on OKD is still the initial [version (v0.1.0)](https://github.com/openshift/aws-pod-identity-webhook/tree/v0.1.0) which currently cannot set these parameters, because of [a bug](https://github.com/aws/amazon-eks-pod-identity-webhook/pull/120).  Regionality should be either set true by default or the operator should be updated to a version that allows env injection w/ sa annotation.

Our requirement is to be able to restrict cross-account IRSA conditionally and utilize an interface endpoint for regional sts.
**Version**
4.13.0-0.okd-2023-05-22-052007

**How reproducible**
Annotation `eks.amazonaws.com/sts-regional-endpoints: "true"` injects env `AWS_STS_REGIONAL_ENDPOINTS=regional` into the IRSA pod.

**Log bundle**
N/A

**Comment by vrutkovs** [comment]
>Outdated pod-identity-webhook version

Is it part of OKD payload or installed as additional operator from community OperatorHub?

**Reply by matveidner** [reply]
It is default, as part of the cloud-credential ClusterOperator

**Reply by vrutkovs** [reply]
OKD uses the same cloud-credential operator as OCP. So this issue should be reported as OCP bug, fixed there, backported to 4.13 - and it would land in OKD 4.13 stable
